src/corelibs/U2Formats/src/TextDocumentFormat.h

```diff
@@ -41,6 +49,61 @@ class U2FORMATS_EXPORT TextDocumentFormat : public DocumentFormat {
     virtual Document *loadTextDocument(IOAdapter *io, const U2DbiRef &dbiRef, const QVariantMap &fs, U2OpStatus &os) = 0;
 };
 
+/*
+ * Base class for all text (non-binary) document formats that can be opened in a usual text editor.
+ */
+class U2FORMATS_EXPORT TextDocumentFormat : public DocumentFormat {
+public:
+    TextDocumentFormat(QObject *p, const DocumentFormatId &id, DocumentFormatFlags _flags, const QStringList &fileExts = QStringList());
+
+protected:
+    /** Checks if the 'rawTextData' text is stored using this document format. */
+    virtual FormatCheckResult checkRawTextData(const QString &rawTextData, const GUrl &originalDataUrl) const = 0;
+
+    /**
+     * Loads a document from the given stream.
+     * 'hints' map contains set of loading hints like 'DocumentReadingMode_SequenceMergeGapSize', etc…
+     * If loading is failed the nullptr is returned and the error is stored in 'os', otherwise a non-nullptr value is returned.
+     */
+    virtual Document *loadTextDocument(TextStreamReader &reader, const U2DbiRef &dbiRef, const QVariantMap &hints, U2OpStatus &os) = 0;
+
+    /**
+     * Loads a DNASequence object from the current state of the reader.
+     * Returns a valid sequence object or sets an error to the 'os'.
+     * By default returns nullptr & error that sequence loading is not supported by the current format.
+     */
+    virtual DNASequence *loadTextSequence(TextStreamReader &reader, U2OpStatus &os);
```

Comment:
Why isn't this method pure virtual like `loadTextDocument`?

Reply:
It has a default impl that is reusable for all text formats that do not have DNASequence objects inside.
The loadTextDocument is the 'must' method to override for all text document formats, so it is pure virtual.